packages/components/src/ErrorView.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useCallback, useLayoutEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { vsDiffAdded, vsDiffRemoved, vsWarning } from '@deephaven/icons';
+import {
+  useDebouncedCallback,
+  useResizeObserver,
+} from '@deephaven/react-hooks';
+import CopyButton from './CopyButton';
+import Button from './Button';
+import './ErrorView.scss';
+
+export type ErrorViewerProps = {
+  /** The message to display in the error view */
+  message: string;
+
+  /** Set to true if you want the error view to display expanded. Will not show the Show More/Less buttons if true. Defaults to false. */
+  isExpanded?: boolean;
+
+  /** The type of error message to display in the header. Defaults to Error. */
+  type?: string;
+};
+
+/**
+ * Component that displays an error message in a textarea so user can scroll and a copy button.
+ */
+function ErrorView({
+  message,
+  isExpanded: isExpandedProp = false,
+  type = 'Error',
+}: ErrorViewerProps): JSX.Element {
+  const [isExpandable, setIsExpandable] = useState(false);
+  const [isExpanded, setIsExpanded] = useState(false);
+  const viewRef = useRef<HTMLDivElement>(null);
+  const textareaRef = useRef<HTMLPreElement>(null);
+
+  const handleResize = useCallback(() => {
+    if (isExpanded || textareaRef.current == null) {
```

Comment:
should also check `isExpandedProp` ?

Reply:
Doesn't need to, `isExpandedProp` being true will never show the expand more/less button. Don't need to change this logic.

Reply:
Yeah but aren't you doing an unnecessary work still? Checking height can trigger a forced reflow.